pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -759,8 +759,14 @@ type Address struct {
 
 //END Video -------------------
 
+type Stats struct {
+	Period int `xml:"period,attr"`
```

Comment:
I am not sure if we want to make this configurable by users.
@andreabolognani do you know about sources which wound indicate reasonable values for it? Would be great if you could share some insights.

Reply:
Sorry, I don't have any specific insights into what period could be a good default. The obvious answer would be to look at what OpenStack and oVirt are doing, though I'm not sure you'd find that a very satisfying one :)

I think I share your opinion that this is not necessarily something that you need to expose to users, at least not right away: if the goal is to have ballooning stats available, then picking a reasonable, static default might the the way to go.

Reply:
Just note we are exposing this https://github.com/kubevirt/kubevirt/pull/3568/files#diff-3bfc6e1b38439eeb33078da0a69ef239R378. I take period=10 as an example from OpenStack. (Couldn't find better...) In my understanding, if VM will be swapping for 9sec and we collect it on the 10th we will not capture this. Therefore I suggested to expose it as users can find a suitable value for their workloads.

Also, note that we collect metrics only when the Prometheus registry does... (so you can miss swapping entirely) but I might be wrong here...